snapshot/policy/retention_policy.go

```diff
@@ -145,25 +148,35 @@ func (r *RetentionPolicy) getRetentionReasons(i int, s *snapshot.Manifest, cutof
 		{cutoff.weekly, fmt.Sprintf("%04v-%02v", yyyy, wk), "weekly", r.KeepWeekly},
 		{cutoff.daily, s.StartTime.Format("2006-01-02"), "daily", r.KeepDaily},
 		{cutoff.hourly, s.StartTime.Format("2006-01-02 15"), "hourly", r.KeepHourly},
+		{cutoff.within, s.StartTime.Format("2006-01-02 15"), "within", r.KeepWithin},
```

Comment:
i don't think this logic belongs here, you should have a separate for loop, this will avoid a bunch of hacks in the current loop.

Reply:
What changes do I need to make

Reply:
I've left some suggestions below. Alternatively, (and probably more correctly), this logic should be pulled out of this loop/function completely. See #660 for a similar change

Reply:
I have removed this logic from loop.